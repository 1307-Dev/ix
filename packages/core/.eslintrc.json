{
  "root": true,
  "extends": [
      "plugin:@stencil/recommended"
  ],
  "rules": {
    "react/jsx-no-bind": 0,
    "@stencil/async-methods": 0,
    "@stencil/strict-boolean-conditions": 0
  }
}
