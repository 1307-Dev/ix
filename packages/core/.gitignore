dist/
www/
loader/

*~
*.sw[mnpcod]
*.log
*.lock
*.tmp
*.tmp.*
log.txt
*.sublime-project
*.sublime-workspace

component-doc.json
.stencil/
.idea/
.vscode/
.sass-cache/
.versions/
node_modules/
$RECYCLE.BIN/

.DS_Store
Thumbs.db
UserInterfaceState.xcuserstate
.env
coverage
/test-results/
/playwright-report/
/playwright/.cache/


/ix-brand-theme
