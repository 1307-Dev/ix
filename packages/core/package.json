{
  "name": "@siemens/ix",
  "version": "0.0.0",
  "license": "Siemens Inner Source License v1.4",
  "description": "Stencil Component Starter",
  "main": "dist/index.cjs.js",
  "module": "dist/index.js",
  "es2015": "dist/esm/index.mjs",
  "es2017": "dist/esm/index.mjs",
  "types": "dist/types/index.d.ts",
  "collection": "dist/collection/collection-manifest.json",
  "collection:main": "dist/collection/index.js",
  "unpkg": "dist/core-ui-core/core-ui-core.js",
  "files": [
    "dist/",
    "loader/",
    "src/**/*.md",
    "scss/"
  ],
  "scripts": {
    "build": "stencil build --docs --prod",
    "build:dev": "stencil build --docs",
    "lint": "eslint src/**/*{.ts,.tsx}",
    "start": "stencil build --dev --watch --serve",
    "dev": "stencil build --watch",
    "test": "stencil test --spec",
    "test.watch": "stencil test --spec --watchAll --no-cache",
    "test.coverage": "stencil test --spec --coverage",
    "test.e2e": "playwright test --reporter html",
    "test.e2e.list": "playwright test --reporter list",
    "host-root": "http-server ./ -a 127.0.0.1 -p 8080",
    "generate": "stencil generate",
    "postversion": "echo \"/*\n * COPYRIGHT (c) Siemens AG 2018-2022 ALL RIGHTS RESERVED.\n */\nexport const version = '$npm_package_version'\" > ./src/env.ts",
    "generate.theme": "ts-node -P ./scripts/tsconfig.json ./scripts/generate-theme"
  },
  "dependencies": {
    "@siemens/core-ui-icons": "~1.1.16",
    "@stencil/core": "~2.15.0",
    "animejs": "~3.2.1",
    "hyperlist": "^1.0.0"
  },
  "devDependencies": {
    "@playwright/test": "^1.24.1",
    "@popperjs/core": "~2.11.0",
    "@siemens/core-ui-icons": "^1.1.16",
    "@stencil/angular-output-target": "^0.4.0",
    "@stencil/eslint-plugin": "^0.4.0",
    "@stencil/postcss": "^2.0.0",
    "@stencil/react-output-target": "^0.3.1",
    "@stencil/sass": "~1.5.2",
    "@testing-library/dom": "^8.11.0",
    "@types/animejs": "^3.1.4",
    "@types/jest": "^26.0.18",
    "@types/mustache": "^4.1.1",
    "@types/puppeteer": "^5.4.0",
    "@typescript-eslint/eslint-plugin": "^4.33.0",
    "@typescript-eslint/parser": "^4.33.0",
    "animate.css": "~4.1.1",
    "autoprefixer": "10.4.5",
    "bootstrap": "~5.2.0",
    "cz-conventional-changelog": "^3.3.0",
    "eslint": "^7.32.0",
    "eslint-plugin-react": "^7.26.1",
    "fs-extra": "^10.0.0",
    "http-server": "^14.1.1",
    "jest": "^26.6.3",
    "jest-cli": "^26.6.3",
    "mustache": "^4.2.0",
    "puppeteer": "^10.0.0",
    "sass": "^1.54.0",
    "start-server-and-test": "^1.14.0",
    "ts-node": "^10.0.0",
    "typescript": "^4.2.4"
  },
  "peerDependencies": {
    "@popperjs/core": "^2.11.0",
    "bootstrap": "~5.2.0"
  },
  "publishConfig": {
    "registry": "***REMOVED***"
  },
  "config": {
    "commitizen": {
      "path": "./node_modules/cz-conventional-changelog"
    }
  }
}
