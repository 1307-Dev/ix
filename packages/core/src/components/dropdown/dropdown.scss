@import '~bootstrap/scss/functions';
@import 'common-variables';
@import '~bootstrap/scss/variables';
@import '~bootstrap/scss/mixins';
@import '~bootstrap/scss/dropdown';

@import 'mixins/fonts';
@import 'mixins/text-truncation';
@import 'components/dropdown';

:host {
  min-width: 0px;
}
