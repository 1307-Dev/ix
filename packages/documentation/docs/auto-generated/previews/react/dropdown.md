<!--
SPDX-FileCopyrightText: 2022 Siemens AG

SPDX-License-Identifier: MIT
-->

<!--
* COPYRIGHT (c) Siemens AG
*
* This source code is licensed under the MIT license found in the
* LICENSE file in the root directory of this source tree.
-->
  
<!-- Auto generated! Please edit here: siemens-ix/packages/react-test-app/src/preview-examples/dropdown.tsx -->
```tsx
import { IxButton, IxDropdown, IxDropdownItem } from '@siemens/ix-react';
import React from 'react';

export const Dropdown: React.FC = () => {
  return (
    <>
      <IxButton id="triggerId">Open</IxButton>
      <IxDropdown trigger="triggerId">
        <IxDropdownItem label="Item 1"></IxDropdownItem>
        <IxDropdownItem label="Item 2"></IxDropdownItem>
        <IxDropdownItem label="Item 3"></IxDropdownItem>
      </IxDropdown>
    </>
  );
};
```
