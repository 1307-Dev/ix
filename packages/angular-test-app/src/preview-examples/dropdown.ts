import { Component } from '@angular/core';

@Component({
  selector: 'app-dropdown',
  template: `
    <ix-button id="triggerId">Open</ix-button>
    <ix-dropdown trigger="triggerId">
      <ix-dropdown-item label="Item 1"></ix-dropdown-item>
      <ix-dropdown-item label="Item 2"></ix-dropdown-item>
      <ix-dropdown-item label="Item 3"></ix-dropdown-item>
    </ix-dropdown>
  `,
})
export class Dropdown {}
