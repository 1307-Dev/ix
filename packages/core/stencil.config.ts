import { angularOutputTarget } from '@stencil/angular-output-target';
import { Config } from '@stencil/core';
import { postcss } from '@stencil/postcss';
import { reactOutputTarget } from '@stencil/react-output-target';
import { sass } from '@stencil/sass';
import autoprefixer from 'autoprefixer';
export const config: Config = {
  bundles: [
    {
      components: ['cw-animated-tabs', 'cw-animated-tab'],
    },
    {
      components: ['cw-application-header'],
    },
    {
      components: ['cw-basic-navigation'],
    },
    {
      components: ['cw-blind'],
    },
    {
      components: ['cw-breadcrumb', 'cw-breadcrumb-item'],
    },
    {
      components: ['cw-button'],
    },
    {
      components: ['cw-category-filter'],
    },
    {
      components: ['cw-chip'],
    },
    {
      components: ['cw-counter-pill'],
    },
    {
      components: ['cw-drawer'],
    },
    {
      components: ['cw-dropdown', 'cw-dropdown-item'],
    },
    {
      components: ['cw-event-list', 'cw-event-list-item'],
    },
    {
      components: ['cw-expanding-search'],
    },
    {
      components: ['cw-filter-chip'],
    },
    {
      components: ['cw-flip-tile', 'cw-flip-tile-content'],
    },
    {
      components: ['cw-group', 'cw-group-dropdown-item', 'cw-group-item'],
    },
    {
      components: ['cw-icon'],
    },
    {
      components: ['cw-icon-button'],
    },
    {
      components: ['cw-input-group'],
    },
    {
      components: ['cw-map-navigation', 'cw-map-navigation-overlay'],
    },
    {
      components: [
        'cw-menu',
        'cw-menu-item',
        'cw-menu-about',
        'cw-menu-about-item',
        'cw-menu-about-news',
        'cw-menu-avatar',
        'cw-menu-avatar-item',
        'cw-menu-settings',
        'cw-menu-settings-item',
      ],
    },
    {
      components: ['cw-message-bar'],
    },
    {
      components: ['cw-modal', 'cw-modal-container'],
    },
    {
      components: ['cw-pill'],
    },
    {
      components: ['cw-select', 'cw-select-item'],
    },
    {
      components: ['cw-spinner'],
    },
    {
      components: ['cw-split-button', 'cw-split-button-item'],
    },
    {
      components: ['cw-tile'],
    },
    {
      components: ['cw-toast', 'cw-toast-container'],
    },
    {
      components: ['cw-toggle'],
    },
    {
      components: ['cw-tree', 'cw-tree-item'],
    },
    {
      components: ['cw-upload'],
    },
  ],
  extras: {
    appendChildSlotFix: true,
    slotChildNodesFix: true,
  },
  namespace: 'core-ui-core',
  globalStyle: './scss/v7/core-ui.scss',
  // globalStyle: 'scss/core-ui.scss',
  minifyCss: false,
  plugins: [
    sass({
      includePaths: ['./scss/v7', './scss'],
    }),
    postcss({
      plugins: [autoprefixer()],
    }),
  ],
  outputTargets: [
    angularOutputTarget({
      componentCorePackage: '@siemens/ix',
      directivesProxyFile: '../angular/src/components.ts',
      directivesArrayFile: '../angular/src/declare-components.ts',
      excludeComponents: ['my-component'],
    }),
    reactOutputTarget({
      componentCorePackage: '@siemens/ix',
      proxiesFile: '../react/src/components.ts',
      includeDefineCustomElements: true,
      excludeComponents: ['my-component'],
    }),
    {
      type: 'dist',
      esmLoaderPath: '../loader',
      copy: [
        {
          src: '../scss/fonts',
          dest: '../core-ui-core/fonts',
        },
      ],
    },
    {
      type: 'dist-custom-elements',
    },
    {
      type: 'docs-json',
      file: 'component-doc.json',
    },
    {
      type: 'docs-readme',
      footer: '',
      dependencies: false,
    },
    {
      type: 'www',
      serviceWorker: null, // disable service workers
      copy: [
        {
          src: './../../../node_modules/@siemens/core-ui-icons/dist',
          dest: 'build/core-ui-webfont',
        },
        {
          src: '../scss/fonts',
          dest: 'build/fonts',
        },
      ],
    },
  ],
};
