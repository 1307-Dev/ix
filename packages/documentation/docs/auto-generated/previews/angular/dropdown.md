<!-- Auto generated! Please edit here: siemens-ix/packages/angular-test-app/src/preview-examples/dropdown.ts -->
```typescript
/*
 * COPYRIGHT (c) Siemens AG 2018-2022 ALL RIGHTS RESERVED.
 */
import { Component } from '@angular/core';

@Component({
  selector: 'app-dropdown',
  template: `
    <ix-button id="triggerId">Open</ix-button>
    <ix-dropdown trigger="triggerId">
      <ix-dropdown-item label="Item 1"></ix-dropdown-item>
      <ix-dropdown-item label="Item 2"></ix-dropdown-item>
      <ix-dropdown-item label="Item 3"></ix-dropdown-item>
    </ix-dropdown>
  `,
})
export class Dropdown {}
```
