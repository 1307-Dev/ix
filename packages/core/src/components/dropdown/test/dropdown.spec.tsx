/*
 * COPYRIGHT (c) Siemens AG 2018-2022 ALL RIGHTS RESERVED.
 */

import { newSpecPage } from '@stencil/core/testing';
import { Dropdown } from '../dropdown';
import { fireEvent } from '@testing-library/dom';

describe('cw-dropdown', () => {
  let page: any;
  let dropdown: any;

  beforeEach(async () => {
    page = await newSpecPage({
      components: [Dropdown],
      html: `
      <cw-dropdown>
        <cw-dropdown-item></cw-dropdown-item>
      </cw-dropdown>
      `,
    });

    dropdown = document.querySelector('cw-dropdown');
  });

  it('renders', async () => {
    expect(page.root).toMatchSnapshot();
  });

  it('should open with anchor element', async () => {
    const page = await newSpecPage({
      components: [Dropdown],
      html: `<div></div>`,
    });

    const anchor = page.doc.createElement('cw-button');
    anchor.setAttribute('id', 'test');
    anchor.innerText = 'Test';

    const dropdown = page.doc.createElement('cw-dropdown');
    dropdown.show = true;

    const dropdownItem = page.doc.createElement('cw-dropdown-item');

    dropdown.appendChild(dropdownItem);

    page.root.appendChild(anchor);
    page.root.appendChild(dropdown);

    await page.waitForChanges();

    expect(page.root).toEqualHtml(`
    <cw-dropdown class="dropdown-menu show" show="" style="margin: 0; min-width: 0px;">
      <div style="display: contents;">
        <cw-dropdown-item></cw-dropdown-item>
      </div>
    </cw-dropdown>
    `);
  });

  it('should collapse, when clicked outside', async () => {
    dropdown.show = true;
    await page.waitForChanges();

    fireEvent.click(window);  //click outside
    await page.waitForChanges();

    expect(dropdown.show).toBeFalsy();
  });

  it('emits an event, when show changed', async () => {
    const mockCallback = jest.fn();
    page.win.addEventListener('showChanged', mockCallback);

    dropdown.show = true;
    await page.waitForChanges();

    fireEvent.click(window);  
    await page.waitForChanges(); 
    expect(mockCallback).toHaveBeenCalled();
  });

});