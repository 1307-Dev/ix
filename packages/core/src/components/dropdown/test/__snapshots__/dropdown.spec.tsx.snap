// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cw-dropdown renders 1`] = `
<cw-dropdown class="dropdown-menu" style="margin: 0; min-width: 0px;">
  <!---->
  <div style="display: contents;">
    <cw-dropdown-item></cw-dropdown-item>
  </div>
</cw-dropdown>
`;
